src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateResolver.cs

```diff
@@ -485,5 +443,197 @@ private static void AddParameterMatchingToTemplates(IReadOnlyCollection<ITemplat
+        /// The class is used for template filtering using specific <see cref="CliNameFilter(string)"/> filter which takes into account the short names of template group when matching names.
+        /// </summary>
+        private class TemplateInfoWithGroupShortNames : ITemplateInfo, IShortNameList
+        {
+            private TemplateInfoWithGroupShortNames (ITemplateInfo source)
+            {
+                Author = source.Author;
+                Description = source.Description;
+                Classifications = source.Classifications;
+                DefaultName = source.DefaultName;
+                Identity = source.Identity;
+                GeneratorId = source.GeneratorId;
+                GroupIdentity = source.GroupIdentity;
+                Precedence = source.Precedence;
+                Name = source.Name;
+                ShortName = source.ShortName;
+                Tags = source.Tags;
+                CacheParameters = source.CacheParameters;
+                Parameters = source.Parameters;
+                MountPointUri = source.MountPointUri;
+                ConfigPlace = source.ConfigPlace;
+                LocaleConfigPlace = source.LocaleConfigPlace;
+                HostConfigPlace = source.HostConfigPlace;
+                ThirdPartyNotices = source.ThirdPartyNotices;
+                BaselineInfo = source.BaselineInfo;
+                HasScriptRunningPostActions = source.HasScriptRunningPostActions;
+            }
+
+            internal static TemplateInfoWithGroupShortNames FromITemplateInfo(ITemplateInfo source)
+            {
+                TemplateInfoWithGroupShortNames filterableTemplate = new TemplateInfoWithGroupShortNames(source);
+
+                if (source is IShortNameList sourceWithShortNameList)
+                {
+                    filterableTemplate.ShortNameList = sourceWithShortNameList.ShortNameList;
+                }
+                return filterableTemplate;
+            }
+
+            public string Author { get; private set; }
+
+            public string Description { get; private set; }
+
+            public IReadOnlyList<string> Classifications { get; private set; }
+
+            public string DefaultName { get; private set; }
+
+            public string Identity { get; private set; }
+
+            public Guid GeneratorId { get; private set; }
+
+            public string GroupIdentity { get; private set; }
+
+            public int Precedence { get; private set; }
+
+            public string Name { get; private set; }
+
+            public string ShortName { get; private set; }
+
+            public IReadOnlyList<string> ShortNameList { get; private set; } = new List<string>();
+
+            public IReadOnlyList<string> GroupShortNameList { get; set; } = new List<string>();
```

Comment:
setter on this can become private, and `groupShortNameList` could be passed as 2nd parameter to constructor.

Reply:
I also wonder, if it would be OK, to remove this property and override `ShortNameList` with `GroupShortNameList` value.

Reply:
> I also wonder, if it would be OK, to remove this property and override `ShortNameList` with `GroupShortNameList` value.

The first version did exactly this, but then i preferred to kept them separate and do specific filter for CLI to manage it.